com.unity.netcode.gameobjects/Runtime/Spawning/NetworkSpawnManager.cs

```diff
@@ -668,17 +676,21 @@ internal void OnDespawnObject(NetworkObject networkObject, bool destroyGameObjec
                 return;
             }
 
-            // Move child NetworkObjects to the root when parent NetworkObject is destroyed
-            foreach (var spawnedNetObj in SpawnedObjectsList)
+            // If we are shutting down the NetworkManager, then ignore resetting the parent
```

Comment:
@MFatihMAR 
This was my solution to handle the issue where changing the parent would cause NetworkObject.OnTransformParentChanged to log an exception because NetworkManager is null during shutdown when the NetworkObjects are being despawned.  Adding an additional series of checks in NetworkObject.OnTransformParentChanged seemed less-than-optimal since this is specific to only despawning.

Reply:
```cs
if (NetworkManager == null || !NetworkManager.IsListening)
{
    transform.parent = m_CachedParent;
    Debug.LogException(new NotListeningException($"{nameof(NetworkManager)} is not listening, start a server or host before reparenting"));
    return;
}
```

is this the check you're referring to?

if so, we could split that logic up into 2 pieces: `NetworkManager == null` and `NetworkManager.IsListening`.
we'd return early if `NetworkManager == null`
we'd log exception if `!NetworkManager.IsListening`

also, I think we should not introduce the `NetworkManager.ShutdownInProgress` (see my other comment above) and also not inject this extra check here.